ckan/views/group.py

```diff
@@ -0,0 +1,1160 @@
+    group_type = c.group_dict['type']
+    _setup_template_variables(context, {'id': id}, group_type=group_type)
+    return base.render(
+        _about_template(group_type), extra_vars={'group_type': group_type})
+
+
+def members(id):
+    group_type = _guess_group_type()
+
+    context = {'model': model, 'session': model.Session, 'user': c.user}
+
+    try:
+        data_dict = {'id': id}
+        check_access('group_edit_permissions', context, data_dict)
+        c.members = get_action('member_list')(context, {
+            'id': id,
+            'object_type': 'user'
+        })
+        data_dict['include_datasets'] = False
+        c.group_dict = _action('group_show')(context, data_dict)
+    except NotFound:
+        base.abort(404, _('Group not found'))
+    except NotAuthorized:
+        base.abort(403,
+                   _('User %r not authorized to edit members of %s') % (c.user,
+                                                                        id))
+
+    return _render_template('group/members.html', group_type)
+
+
+def member_delete(id):
+    group_type = _guess_group_type()
+
+    if 'cancel' in request.params:
+        return _redirect_to_this_controller(action='members', id=id)
+
+    context = {'model': model, 'session': model.Session, 'user': c.user}
+
+    try:
+        _check_access('group_member_delete', context, {'id': id})
+    except NotAuthorized:
+        base.abort(403, _('Unauthorized to delete group %s members') % '')
+
+    try:
+        user_id = request.params.get('user')
+        if request.method == 'POST':
+            _action('group_member_delete')(context, {
+                'id': id,
+                'user_id': user_id
+            })
+            h.flash_notice(_('Group member has been deleted.'))
+            return _redirect_to_this_controller(action='members', id=id)
+        c.user_dict = _action('group_show')(context, {
+            'id': user_id
+        })
+        c.user_id = user_id
+        c.group_id = id
+    except NotAuthorized:
+        base.abort(403, _('Unauthorized to delete group %s members') % '')
+    except NotFound:
+        base.abort(404, _('Group not found'))
+    return _render_template('group/confirm_delete_member.html', group_type)
+
+
+def history(id):
+    group_type = _guess_group_type()
+    if 'diff' in request.params or 'selected1' in request.params:
+        try:
+            params = {
+                'id': request.params.getone('group_name'),
+                'diff': request.params.getone('selected1'),
+                'oldid': request.params.getone('selected2'),
+            }
+        except KeyError:
+            if 'group_name' in dict(request.params):
+                id = request.params.getone('group_name')
+            c.error = \
+                _('Select two revisions before doing the comparison.')
+        else:
+            params['diff_entity'] = 'group'
+            h.redirect_to(controller='revision', action='diff', **params)
```

Comment:
I believe that flask generates responses instead of exceptions(like pylons) for redirect, so you need to put return at this line. Same for https://github.com/ckan/ckan/pull/4007/files#diff-244c9c0a85f9bb077ce1799c1af6210cR1032

Reply:
i was rebasing my branch i i think it got corrupted, some of the changes you mention, are already done